var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -56,15 +59,25 @@ def edit(self, spec, prefix):
             spec.variants['bsws'].value)
 
     def build(self, spec, prefix):
-        make("shared")
+        with working_dir('src'):
+            if '~shared' in spec:
+                make('static')
+            else:
+                make('shared')
 
     def install(self, spec, prefix):
         incdir = 'include' if spec.satisfies('@0.5.1:') else 'inc'
 
+        if self.run_tests:
+            make('test')
+
         # No install provided
         mkdirp(prefix.lib)
         mkdirp(prefix.include)
-        install('lib/libzfp.so', prefix.lib)
+        if '~shared' in spec:
+            install('lib/libzfp.a', prefix.lib)
+        else:
+            install('lib/libzfp.so', prefix.lib)
```

Comment:
Use `dso_suffix`. It will work on both Linux and macOS.

Reply:
I use it on osx. it works. The package doesn't build .dylib files. It builds .sos. So, that is what I went with.

Reply:
wow, hopefully other package which depend on it know what to do... Maybe add a incode comment above this line saying that it builds `.so` by default on macOS as well.

Reply:
Ok, wait, the way zfp package is coded now, the instructions to actuall perform the installation steps of putting the lib and header files in the install point will _not_ get executed if `make('test')` fails. But, if I remove that code from the package and rely upon Spack to do it. How will it _avoid_ doing the actual installation if `make('test')` fails? Does it do the test *after* the install and then uninstall if the test fails?

Reply:
Ok, we;ll, I think I see it. It will do the `build` phase, then call `make('test')` and then not go on to install phase if test passes